blockchain/pallets/minting/src/lib.rs

```diff
@@ -190,9 +190,18 @@ pub mod pallet {
             }
 
             IdDatasets::<T>::insert(id, extension_proof);
-            NextMintingRewards::<T>::insert(id, who);
 
-            Ok(())
+            // Only first call is free, determined by presence of
+            // entry in NextMintingRewards. This entry will be removed when
+            // minting rewards get dispersed in the on_finalize block. Not present
+            // indicates its the first call.
+            if NextMintingRewards::<T>::try_get(id).is_ok() {
+                // None indicates use pallet::weight annotation
+                Ok(None.into())
+            } else {
+                NextMintingRewards::<T>::insert(id, who);
+                Ok(Some(0).into())
+            }
```

Comment:
With this logic, I'd expect the following test to fail:

register_identity
register_for_minting

run_to_minting
let dispatch_info = register_for_minting
assert!(dispatch_info.pays_fee == Pays::No)

Reply:
Interesting, I've added the test and you are right, it does not pass. 

```
for (id, account) in recipients { ‣id: &u64 ‣account: &<T as Config>::AccountId
   T::Currency::deposit_creating(account, actual_mint_per_user);
   NextMintingRewards::<T>::remove(id);
}
```

my assumption was that the `NextMintingRewards::<T>::remove(id)` in the `on_finalize` would remove the entry in `NextMintingRewards`. What am I missing here?

Reply:
This is resolved after our conversation, the idea is that only the first call ever should be free, my understanding was that the first call in each "epoch" should be free.